ping.go

```diff
@@ -295,27 +295,26 @@ func (p *Pinger) Privileged() bool {
 // Run runs the pinger. This is a blocking function that will exit when it's
 // done. If Count or Interval are not specified, it will run continuously until
 // it is interrupted.
-func (p *Pinger) Run() {
+func (p *Pinger) Run() (err error) {
```

Comment:
Hrm, I'm not sure what is the best Go style here. We're sometimes using named, and sometimes using un-named returns.

Reply:
I was wondering about this as well. Is there a more idiomatic go way?

Reply:
IIRC both are the idiomatic way. I'm using named return in this case so less code needs to be changed because the original code declares a `var err error` in the scope.

Reply:
From https://tour.golang.org/basics/7

> Naked return statements should be used only in short functions, as with the example shown here. They can harm readability in longer functions

I think `Run()` qualifies as long enough that we should use explicit `return err`, rather than naked `return`s.